node/src/main/java/io/airlift/node/NodeConfig.java

```diff
@@ -169,4 +164,21 @@ public NodeConfig setConfigSpec(String configSpec)
         this.configSpec = configSpec;
         return this;
     }
+
+    public AddressType getAddressType()
+    {
+        return addressType;
+    }
+
+    @Config("node.internal-address-type")
```

Comment:
You can specify node internal adress manually, with the `node.internal-address`. 
It this property is not specified, framework resolves the local address automatically. There are 3 types of automatically resolved adress. It can be a bare `IP` resolved by the existing algoritm. It can be a hostname. And it can be a FQDN.